server/src/main/java/org/elasticsearch/cluster/metadata/MappingMetadata.java

```diff
@@ -42,10 +43,13 @@ public class MappingMetadata implements SimpleDiffable<MappingMetadata> {
 
     private final boolean routingRequired;
 
+    private final Map<String, Set<String>> fieldsForModels;
+
     public MappingMetadata(DocumentMapper docMapper) {
         this.type = docMapper.type();
         this.source = docMapper.mappingSource();
         this.routingRequired = docMapper.routingFieldMapper().required();
+        this.fieldsForModels = null;  // TODO: Set fieldsForModels here
```

Comment:
This is the main connection between IndexMetadata and the actual field mappings - using the `MappingLookup` for retrieving information from mappings.

The other constructors don't have enough info for creating this field.